### DISCUSSION TITLE
I2C driver used by multiple components

### DISCUSSION THREAD

**Discussion by Gonta01** [nasa/fprime#2850] [Q&A]
How to properly organize topology that has I2C driver used by multiple components? 
[I2cDriverPorts](https://github.com/nasa/fprime/tree/devel/Drv/I2cDriverPorts) implies using of Fw.Buffer. Does it mean that I2C driver also must have allocate and deallocate ports? [LinuxI2cDriver](https://github.com/nasa/fprime/blob/devel/Drv/LinuxI2cDriver/LinuxI2cDriver.fpp) doesn't have allocate/deallocate ports.
Are there any reference examples for I2C driver usage?

**Comment by LeStarch** [comment]
Typically, with I2cDriver we wrap an Fw::Buffer around static memory:

```
U8 buffer[size];
buffer[0] = address
buffer[1] = ...
...

Fw::Buffer sendBuffer(buffer, sizeof buffer);

this->writeToI2CPort(0, sendBuffer);
```

Since the LinuxI2CDriver ports are `sync` this works, and we don't need a full allocation/deallocation pattern.  However, you must be careful of the lifetime of the `buffer`.  This is why we are ensuring `sync` ports.

**Reply by Gonta01** [reply]
1) As far as I understand this approach doesn't imply using of StaticMemory or BufferManager components?
2) Could buffer[size] be created locally not statically?
3) "However, you must be careful of the lifetime of the buffer. This is why we are ensuring sync ports."
If possible could you please tell more about consequences if sync port is not used?
